src/grape_block.rs

```diff
@@ -0,0 +1,143 @@
+//! Implements the GrapeBlock, a monitor wrapper for each CyberGrape
+//! processing module. Enforces a common interface between module such that
+//! each module can consume data from the preceding module, process it, and
+//! pass new data to the subsequent module in the CyberGrape pipeline.
+
+use crate::Component;
+use log::{info, warn};
+use std::fmt;
+use std::sync::mpsc::{channel, Receiver, Sender};
+
+///
+/// Runs on its own thread. On receiving data of type A on input, the
+/// GrapeBlock converts them to data of type B, and sends it to its output
+/// channel.
+///
+pub struct GrapeBlock<'a, A, B> {
+    // component is an object of trait Component, and Rust cannot know
+    // how big component is at runtime. any 'dynamically typed' trait
+    // object must be marked with the `dyn` keyboard, post 2021
+
+    // the Box ensures that this GrapeBlock owns the heap-alloc'd ref to the
+    // underlying Component
+
+    // the lifetime 'a indicates that the inner Component cannot live longer
+    // than the GrapeBlock!
+    component: Box<dyn Component<InData = A, OutData = B> + 'a>,
+    output: Sender<B>,
+    input: Receiver<A>,
+}
+
+impl<'a, I, O> GrapeBlock<'a, I, O> {
+    /// Instantiates a new GrapeBlock. The generic types on the GrapeBlock
+    /// enforce that a GrapeBlock's Receiver type must match the InData
+    /// type of its associated Component; and, that its Sender type must
+    /// match the OutData type of its associated Component.
+    pub fn new(
+        component: Box<dyn Component<InData = I, OutData = O> + 'a>,
+        output: Sender<O>,
+        input: Receiver<I>,
+    ) -> Self {
+        env_logger::init();
```

Comment:
`env_logger` should be initialized at the application level, not here in the library, so this line should be removed. A client application may want to use their own logging facade.